### DISCUSSION TITLE
Some bugs for LDR

### DISCUSSION THREAD

**Discussion by BWLIN123** [e0404/matRad#806] [General]
When I ran the original code "matRad_example15_brachy.m" , I encounted one problem, as the fowllowing attached figure shows
![微信截图_20241230164820](https://github.com/user-attachments/assets/4b4a3ffc-d690-4ba5-aa0e-636a76e5d9b9)

So, how to address it?

**Comment by BWLIN123** [comment]
pln.radiationMode   = 'brachy'; 
pln.machine         = 'LDR';    % 'LDR' or 'HDR' for brachy

pln.bioModel        = 'none';
pln.multScen        = 'nomScen';

just to change "HDR" to "LDR", will enount this bug!

**Reply by wahln** [reply]
Thanks for that, will check it out asap and see what the issue is. in the standard script.

**Comment by BWLIN123** [comment]
Great thanks!

Bowen

________________________________
发件人: Niklas Wahl ***@***.***>
发送时间: 2025年1月4日 0:32
收件人: e0404/matRad ***@***.***>
抄送: BWLin ***@***.***>; Author ***@***.***>
主题: Re: [e0404/matRad] Some bugs for LDR (Discussion #806)


Thanks for that, will check it out asap and see what the issue is. in the standard script.

―

**Reply by wahln** [reply]
I fixed this on our [dev](https://github.com/e0404/matRad/tree/dev) branch in this commit: b3649852a0f41a3d5fa33adea0f149d7b40fe975

There was an issue with the shape of the function values to be interpolated. 

Please also consider that the brachytherapy module is highly experimental :wink: 
We are happy for each comments, bug reports, attempts of validation, etc.

**Reply by BWLIN123** [reply]
The code “matRad_example15_brachy.m” runs normally in MATLAB 2024b but encounters errors in MATLAB 2021b and 2022b. The error show in the attached figure.
![Screenshot_20250110101221](https://github.com/user-attachments/assets/5eb5ee1f-39da-487e-95d0-33bc23a805af)

**Reply by wahln** [reply]
there was an out-of-place version check on an unavailable option copied from another optimizer. Should be fixed now on dev
